### PR TITLE
Prefer slices over reference to owned type 

### PR DESCRIPTION
**Pull request by vks**
This fixes two (currently ignored) classes of clippy warnings, and reduces the number of allocations.

### CODE REVIEW

File: game/src/debug/mod.rs [hunk 1]
```diff
@@ -730,7 +730,8 @@ impl ContextualActions for Actions {
                 let center = if pts[0] == *pts.last().unwrap() {
                     // TODO The center looks really wrong for Volunteer Park and others, but I
                     // think it's because they have many points along some edges.
-                    Pt2D::center(&pts.iter().skip(1).cloned().collect())
+                    let v: Vec<_> = pts.iter().skip(1).cloned().collect();
```
Comment (vks) [inline game/src/debug/mod.rs:733]: I could get rid of the intermediate value at the cost of a turbofish.

**Review by dabreegster — CHANGES_REQUESTED** [review]
Wow, this is awesome! I never properly learned how to work with slices, so this is just a bad habit that's persisted over many years. But the point of clippy is to point them out / do really opinionated teaching. Thanks!

File: game/src/debug/mod.rs (hunk 1 above)
Comment (vks) [reply game/src/debug/mod.rs]: > Actually yeah, could you please? Most of the places with all of the intermediate vars are a bit less readable.

Done.

> It's unfortunate that we have to collect at all, but consuming the iterator directly would take way more type declarations at the method definitions than is worth it.

Now that we have `impl Trait`, the type declarations might actually not so bad.

Comment (dabreegster) [reply game/src/debug/mod.rs]: Thanks! I'll keep `impl trait` in mind, but build times are definitely a problem already, so... [turbofish](https://turbo.fish/) is fine.

**Review by dabreegster — APPROVED** [review]
Thanks for the PR! Merging

**Pull request merged**